build/mac/download_hermetic_xcode.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env vpython3
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# Copyright 2018 The Chromium Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+import argparse
+import os
+import platform
+import plistlib
+import subprocess
+import sys
+from urllib.error import URLError  # pylint: disable=no-name-in-module,import-error
+import pkg_resources
+
+import deps
+from deps_config import DEPS_PACKAGES_URL
+
+
+def LoadPList(path):
+    """Loads Plist at |path| and returns it as a dictionary."""
+    with open(path, 'rb') as f:
+        return plistlib.load(f)
+
+
+# This contains binaries from Xcode 13.2.1 13C100, along with the macOS 12 SDK
+XCODE_VERSION = '13.2.1'
+HERMETIC_XCODE_BINARY = (
+    DEPS_PACKAGES_URL +
+    '/xcode-hermetic-toolchain/xcode-hermetic-toolchain-xcode-' +
+    XCODE_VERSION + '-sdk-12.1-12.0.tar.gz')
+
+# The toolchain will not be downloaded if the minimum OS version is not met. 19
+# is the major version number for macOS 10.15. Xcode 13.2 13C90 only runs on
+# 11.3 and newer, but some bots are still running older OS versions. 10.15.4,
+# the OS minimum through Xcode 12.4, still seems to work.
+MAC_MINIMUM_OS_VERSION = [19, 4]
+
+BASE_DIR = os.path.abspath(
+    os.path.join(os.path.dirname(__file__), '..', '..', '..', 'build'))
+TOOLCHAIN_ROOT = os.path.join(BASE_DIR, 'mac_files')
+TOOLCHAIN_BUILD_DIR = os.path.join(TOOLCHAIN_ROOT, 'Xcode.app')
+
+
+def PlatformMeetsHermeticXcodeRequirements():
+    if sys.platform != 'darwin':
+        return False
+    needed = MAC_MINIMUM_OS_VERSION
+    major_version = [
+        int(v) for v in platform.release().split('.')[:len(needed)]
+    ]
+    return major_version >= needed
+
+
+def GetHermeticXcodeVersion(binaries_root):
+    hermetic_xcode_version_plist_path = os.path.join(binaries_root,
+                                                     'Contents/version.plist')
+
+    if not os.path.exists(hermetic_xcode_version_plist_path):
+        return ''
+
+    hermetic_xcode_version_plist = LoadPList(hermetic_xcode_version_plist_path)
+    return hermetic_xcode_version_plist['CFBundleShortVersionString']
+
+
+def InstallXcodeBinaries():
+    """Installs the Xcode binaries and accepts the license."""
+
+    # only download for Brave goma users
+    goma_server_host = os.environ.get('npm_config_goma_server_host')
+    if goma_server_host is None or not goma_server_host.endswith('.brave.com'):
+        print("Goma server host is not configured for Brave")
```

Comment:
`Skipping hermetic Xcode install, goma server host is not configured for Brave`?

Reply:
I think we're going to switch to downloading from github so we can just make it a private repo using git auth

Reply:
https://github.com/brave/xcode-hermetic-toolchain/releases/download/xcode-13.2.1-sdk-12.1-12.0/xcode-hermetic-toolchain.tar.gz